doc/design/cluster_train/save_model.md

```diff
@@ -0,0 +1,97 @@
+# Design Doc: Save Model
+
+## Overview
+
+The model is the output of the training process. There are two
+ways from which user can obtain a model:
+
+- Save model triggered by user code: user code asks PaddlePaddle to
+  save a model.
+- Convert model from the snapshot: model being converted from
+  pservers' periodic snapshot. In this way, the user can cancel a job
+  at any time, and still have a relatively fresh model (we snapshot
+  around every 5 minutes).
+
+### Save Model Triggered by User Code
+
+Both trainers and pservers have access to the model. So the model can
+be saved from a trainer or pservers. We need to decide on where the
+model is saved from.
+
+#### Dense Model vs. Sparse Model
+
+There are two types of model: dense (e.g., weight for a
+fully-connected layer) and sparse model (e.g., word
+embedding). Pservers always jointly have the entire model at any given
```

Comment:
I consult @lcy-seso , she tells me that `sparse model` is a kind of training method to make the model sparse, it's different then `sparse update`. And `language model` and `word emebedding` both have no relation with `sparse model`, and mostly we will not use `sparse model`,  so we can just use `sparse update` in the future.